src/test/java/com/exasol/adapter/dialects/scalarfunction/ScalarFunctionsAbstractIT.java

```diff
@@ -0,0 +1,600 @@
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import com.exasol.adapter.capabilities.ScalarFunctionCapability;
+import com.exasol.errorreporting.ExaError;
+import com.exasol.matcher.TypeMatchMode;
+
+/**
+ * This is an abstract smoke test for all scalar functions, except the geospacial functions (ST_*).
+ */
+@Testcontainers
+@SuppressWarnings("java:S5786") // class is public since it is an abstract test
+@Execution(value = ExecutionMode.SAME_THREAD)
+public abstract class ScalarFunctionsAbstractIT {
+    /**
+     * These have a special syntax, so we define explicit test for them below.
+     */
+    private static final Set<ScalarFunctionCapability> EXCLUDES = Set.of(CASE, CAST, FLOAT_DIV, SESSION_PARAMETER, RAND,
+            ADD, SUB, MULT, NEG, SYS_GUID, SYSTIMESTAMP, CONVERT_TZ, DATE_TRUNC, NUMTODSINTERVAL, NUMTOYMINTERVAL,
+            TO_YMINTERVAL, TO_DSINTERVAL, JSON_VALUE, EXTRACT, POSIX_TIME);
+    private static final String LOCAL_COPY_TABLE_NAME = "LOCAL_COPY";
+    private static final String LOCAL_COPY_SCHEMA = "EXASOL";
+
+    /**
+     * These functions are tested separately in
+     * {@link WithNoParameters#testFunctionsWithNoParenthesis(ScalarFunctionCapability)}
+     */
+    private static final Set<ScalarFunctionCapability> FUNCTIONS_WITH_NO_PARENTHESIS = Set.of(LOCALTIMESTAMP, SYSDATE,
+            CURRENT_SCHEMA, CURRENT_STATEMENT, CURRENT_SESSION, CURRENT_DATE, CURRENT_USER, CURRENT_TIMESTAMP);
+    private static final String LOCAL_COPY_FULL_TABLE_NAME = LOCAL_COPY_SCHEMA + "." + LOCAL_COPY_TABLE_NAME;
+
+    /**
+     * For sum functions we just can guess the parameters because they only work for very specific input. For those we
+     * define explicit parameters here. You can define multiple parameter combinations here (by adding multiple list
+     * entries). By that you can test a function with different values.
+     */
+    private static final Logger LOGGER = LoggerFactory.getLogger(ScalarFunctionsAbstractIT.class);
+    private Set<ScalarFunctionCapability> dialectSpecificExcludes;
+
+    /**
+     * Returns a set of scalar functions that should not be tested for this dialect.
+     * 
+     * @return set of functions
+     */
+    protected abstract Set<ScalarFunctionCapability> getDialectSpecificExcludes();
+
+    /**
+     * Prepare a Virtual Schema with one table that contains columns for the following types: floating point, integer,
+     * string, date, timestamp
+     * <p>
+     * The table must contain exactly one row with examples for each type. Don't use null or 0 as example values.
+     * </p>
+     *
+     * @return the fully qualified name of the table in the virtual schema.
+     */
+    protected abstract String createVirtualSchemaTableWithExamplesForAllDataTypes() throws SQLException;
```

Comment:
I would like to keep it like this, because:
* We can't set the name of the virtual schema table directly. It is inferred by the name of the remote table. That might makes it complex or even impossible to generate virtual schema tables for a given name.
* In a separate PR I would like to change this so that a class like StringTestTable is returned that also has a teardown method.